### PR TITLE
adds HTTP Cookie Storage Monitor

### PR DESCRIPTION
**Pull request by angiemugo**
#### What is the purpose of this pull request?
adds http cookie storage monitor

#### Background
https://github.com/eBardX/XestiMonitors/issues/49

#### Checklist
- [ ] Bug fix (non-breaking change which fixes an issue).
- [X] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to change).
- [X] I have read the **CONTRIBUTING** guidelines.
- [X] My code follows the coding style of this project.
- [X] My change requires a change to the documentation.
- [X] I have documented all my changes.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests pass.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -633,6 +636,7 @@ XestiMonitors is available under [the MIT license][license].
 [focus_monitor]:                                https://eBardX.github.io/XestiMonitors/Classes/FocusMonitor.html
 [gyroscope_monitor]:                            https://eBardX.github.io/XestiMonitors/Classes/GyroscopeMonitor.html
 [heading_monitor]:                              https://eBardX.github.io/XestiMonitors/Classes/HeadingMonitor.html
+[http_cookie_storage_monitor]:          https://eBardX.github.io/XestiMonitors/Classes/HTTPCookieStorageMonitor.html
```
Comment (eBardX) [inline README.md]: Please also update lines 20 and 21.

File: Tests/Foundation/HTTPCookieStorageMonitorTests.swift [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+        if let event = expectedEvent,
+            case let .cookiesChanged(test) = event {
+            XCTAssertEqual(test, httpCookieStorage)
+        } else {
+            XCTFail("Unexpected event")
+        }
+    }
+
+    private func simulate_cookiesAcceptPolicyChanged() {
```
Comment (eBardX) [inline Tests/Foundation/HTTPCookieStorageMonitorTests.swift]: I'd prefer `simulateAcceptPolicyChanged()`.

File: Tests/Foundation/HTTPCookieStorageMonitorTests.swift [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        notificationCenter.post(name: .NSHTTPCookieManagerAcceptPolicyChanged,
+                                object: httpCookieStorage)
+    }
+
+    private func simulate_cookiesChanged() {
```
Comment (eBardX) [inline Tests/Foundation/HTTPCookieStorageMonitorTests.swift]: I'd prefer `simulateCookiesChanged()`.

File: Tests/Foundation/HTTPCookieStorageMonitorTests.swift [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+import XCTest
+@testable import XestiMonitors
+
+internal class HTTPCookiesStorageMonitorTests: XCTestCase {
+    let notificationCenter = MockNotificationCenter()
+    let httpCookieStorage = HTTPCookieStorage()
+
+    override func setUp() {
+        super.setUp()
+        NotificationCenterInjector.inject = { return self.notificationCenter }
```
Comment (eBardX) [inline Tests/Foundation/HTTPCookieStorageMonitorTests.swift]: Please insert a blank line before.

File: Sources/Core/Foundation/HTTPCookieStorageMonitor.swift [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+
+        /// :nodoc:
+        public let rawValue: UInt
+    }
+
+    ///
+    /// Initializes a new `HTTPCookiesStorageMonitor`.
+    ///
+    /// - Parameters:
+    ///   - httpCookieStorage:  The cookie storage to monitor.
```
Comment (eBardX) [inline Sources/Core/Foundation/HTTPCookieStorageMonitor.swift]: Let's call this `cookieStorage` instead (I adjusted the issue wording to match).

File: Sources/Core/Foundation/HTTPCookieStorageMonitor.swift [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+    ///   - options:    The options that specify which events to monitor. By
+    ///                 default, all events are monitored.
+    ///   - queue:      The operation queue on which the handler executes. By
+    ///                 default, the main operation queue is used.
+    ///   - handler:    The handler to call when the text of the
+    ///                 httpCookieStorage changes
+    ///
+    public init(httpCookieStorage: HTTPCookieStorage,
```
Comment (eBardX) [inline Sources/Core/Foundation/HTTPCookieStorageMonitor.swift]: `cookieStorage`

File: Sources/Core/Foundation/HTTPCookieStorageMonitor.swift [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+                options: Options = .all,
+                queue: OperationQueue = .main,
+                handler: @escaping (Event) -> Void) {
+        self.handler = handler
+        self.httpCookieStorage = httpCookieStorage
```
Comment (eBardX) [inline Sources/Core/Foundation/HTTPCookieStorageMonitor.swift]: `cookeStorage`

File: Sources/Core/Foundation/HTTPCookieStorageMonitor.swift [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+        self.options = options
+
+        super.init(queue: .main)
+    }
+
+    ///
+    /// The cookie storage being monitored.
+    ///
+    public let httpCookieStorage: HTTPCookieStorage
```
Comment (eBardX) [inline Sources/Core/Foundation/HTTPCookieStorageMonitor.swift]: `cookieStorage`

File: Tests/Foundation/HTTPCookieStorageMonitorTests.swift [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+//  © 2018 J. G. Pusey (see LICENSE.md)
+//
+
+import Foundation
+import XCTest
+@testable import XestiMonitors
+
+internal class HTTPCookiesStorageMonitorTests: XCTestCase {
+    let notificationCenter = MockNotificationCenter()
+    let httpCookieStorage = HTTPCookieStorage()
```
Comment (eBardX) [inline Tests/Foundation/HTTPCookieStorageMonitorTests.swift]: Let's go with `cookieStorage` here too. (And swap lines 15 and 16 -- I like alphabetical order.)

File: README.md (hunk 1 above)
Comment (angiemugo) [reply README.md]: I am a bit lost here. Is that in the readMe file?

Comment (eBardX) [reply README.md]: My mistake. I meant lines 20 and 21 of `Documents/wrapped-foundation-notifications.md`.

Comment (eBardX) [reply README.md]: Don't worry about it. I will make the final fixes and merge this in.

**Pull request merged**